software/base/src/test/java/brooklyn/entity/brooklynnode/BrooklynNodeIntegrationTest.java

```diff
@@ -361,6 +362,18 @@ protected static void waitForApps(String webConsoleUri) {
         HttpTestUtils.assertHttpStatusCodeEventuallyEquals(webConsoleUri+"/v1/applications", 200);
     }
 
+    protected void waitForApps(final URI webConsoleUri, final int num) {
+        waitForApps(webConsoleUri.toString());
+        
+        // e.g. [{"id":"UnBqPcqg","spec":{"name":"Application (UnBqPcqg)","type":"brooklyn.entity.basic.BasicApplication","locations":["pOL4NtiW"]},"status":"RUNNING","links":{"self":"/v1/applications/UnBqPcqg","entities":"/v1/applications/UnBqPcqg/entities"}}]
+        Asserts.succeedsEventually(new Runnable() {
```

Comment:
Scratch that, the change is for startup (non-rebind) cases only, so good to have.